File: scripts/test_on_off_hbba_subscriber.py
#!/usr/bin/env python3
import rclpy
import rclpy.node

from hbba_lite.msg import Int32Stamped

import hbba_lite


def main():
    rclpy.init()
    node = rclpy.node.Node('test_on_off_hbba_subscriber')

    def callback(data):
        node.get_logger().info('Data received : {}'.format(data.data))

    _ = hbba_lite.OnOffHbbaSubscriber(node, Int32Stamped, 'int_topic_1', callback, 10)

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: src/hbba_lite_cpp_tests/test_on_off_hbba_subscriber.cpp
#include <rclcpp/rclcpp.hpp>

#include <hbba_lite/filters/Subscribers.h>
#include <hbba_lite/msg/int32_stamped.hpp>

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);

    auto node = rclcpp::Node::make_shared("test_on_off_hbba_subscriber");
    OnOffHbbaSubscriber<hbba_lite::msg::Int32Stamped> sub(
        node,
        "int_topic_1",
        10,
        [node](const hbba_lite::msg::Int32Stamped::SharedPtr msg)
        { RCLCPP_INFO(node->get_logger(), "Data received : %i", static_cast<int>(msg->data)); });

    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}
